File: src/java/magma/app/compile/type/PrimitiveRule.java
package magma.app.compile.type;

import magma.api.option.None;
import magma.api.option.Option;
import magma.api.option.Some;
import magma.api.text.Strings;
import magma.app.compile.node.Node;
import magma.app.compile.rule.Rule;

public class PrimitiveRule implements Rule<Node> {
    @Override
    public Option<Node> lex(String input) {
        var stripped = Strings.strip(input);
        if (Primitives.JavaToVariant.containsKey(stripped)) {
            return new Some<Node>(Primitives.JavaToVariant.get(stripped));
        }

        return new None<Node>();
    }
}

File: src/ts/magma/app/compile/type/PrimitiveRule.ts
import { Rule } from "../../../../magma/app/compile/rule/Rule";
import { Node } from "../../../../magma/app/compile/node/Node";
import { Option } from "../../../../magma/api/option/Option";
import { Strings } from "../../../../magma/api/text/Strings";
import { Primitives } from "../../../../magma/app/compile/type/Primitives";
import { Some } from "../../../../magma/api/option/Some";
import { None } from "../../../../magma/api/option/None";
export class PrimitiveRule implements Rule {
	lex(input: string): Option<Node> {
		let stripped = Strings.strip(input)/*unknown*/;
		if (Primitives.JavaToVariant.containsKey(stripped)/*unknown*/){
			return new Some<Node>(Primitives.JavaToVariant.get(stripped))/*unknown*/;
		}
		return new None<Node>()/*unknown*/;
	}
}
